File: src/main/java/com/procurement/access/model/dto/cn/CnProcess.java
package com.procurement.access.model.dto.cn;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.procurement.access.model.dto.ocds.Planning;
import com.procurement.access.model.dto.pn.PnTender;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@JsonPropertyOrder({
        "token",
        "ocid",
        "planning",
        "tender"
})
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class CnProcess {

    @JsonProperty("token")
    private String token;

    @JsonProperty("ocid")
    private String ocId;

    @Valid
    @NotNull
    @JsonProperty("planning")
    private Planning planning;

    @Valid
    @NotNull
    @JsonProperty("tender")
    private CnTender tender;

    @JsonCreator
    public CnProcess(@JsonProperty("token") final String token,
                     @JsonProperty("ocid") final String ocId,
                     @JsonProperty("planning") final Planning planning,
                     @JsonProperty("tender") final CnTender tender) {
        this.token = token;
        this.ocId = ocId;
        this.planning = planning;
        this.tender = tender;
    }
}

File: src/main/kotlin/com/procurement/access/model/dto/cn/CnProcess.kt
package com.procurement.access.model.dto.cn

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.procurement.access.model.dto.ocds.Planning
import javax.validation.Valid
import javax.validation.constraints.NotNull

@JsonInclude(value = JsonInclude.Include.NON_NULL)
data class CnProcess(

        @JsonProperty("ocid")
        var ocid: String?,

        @JsonProperty("token")
        var token: String?,

        @JsonProperty("planning") @Valid @NotNull
        var planning: Planning,

        @JsonProperty("tender") @Valid @NotNull
        var tender: CnTender)
